### ISSUE TITLE
responsibility & communication of plugins

### ISSUE THREAD

**Issue by felixhaeberle** [opral/inlang.com#85]
> Overall, the communication and responsibility of plugins is a mess.
>
> 1. Some do only provide message loading/saving.
> 2. Some do only provide matching functionality for Sherlock.
> 3. Some do both.
>
> 3 is the coolest of them, BUT at the cost of flexibility (can only import from one format), code duplication (same JSON parsing code in every package).<br><br>@samuel.stroschein :<br>The website (cc @nils.jacobsen) just misses a "this plugin defines xyz" section and explainer what x,y, or z do on the product page. I did a sketch a while ago but can't find it anymore.

**Comment by samuelstroschein** [comment]
Yeah, the website should have a "this plugin defines api x, y, z" upon clicking on `x, y, or z`, the docs open with an explainer what this plugin api does

**Comment by NilsJacobsen** [comment]
As a user most of the time, I just want to make my legacy lib compatible with the ecosystem. Therefore I need a  `library-plugins` (next-Intl-plugin, vue-i18n-plugin). That is wasy to understand and comes with all I need to make it work.

**Comment by NilsJacobsen** [comment]
> Yeah, the website should have a "this plugin defines api x, y, z" upon clicking on `x, y, or z`, the docs open with an explainer what this plugin api does

This information should be somewhere, but I think for the most people that will be to complex as the first thing they read.<br><br>What about multiple section on the plugin page. 

1. Library plugins
2. Matcher

They will immediately click on their lib logo, and are there.

**Comment by samuelstroschein** [comment]
@nils.jacobsen your comment is unrelated to the issue. the issue is "what apis does this plugin define?"  

<img src="https://uploads.linear.app/4cb6d418-86ee-4425-b5c8-f91266b0ab28/991a00ee-91b6-4a6e-86b3-19a01aea50fb/228a8fa9-fffe-4c11-9f2d-23fbee151ec5?signature=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJwYXRoIjoiLzRjYjZkNDE4LTg2ZWUtNDQyNS1iNWM4LWY5MTI2NmIwYWIyOC85OTFhMDBlZS05MWI2LTRhNmUtODZiMy0xOWEwMWFlYTUwZmIvMjI4YThmYTktZmZmZS00YzExLTlmMmQtMjNmYmVlMTUxZWM1IiwiaWF0IjoxNzE3NDI0NTk1LCJleHAiOjMzMjg3OTg0NTk1fQ.9Jf_imOawC4VpMF5RSoBIAiRRE8poLrjCKg6xAIMLps " alt="CleanShot 2024-06-03 at 10.22.55@2x.png" width="2208" height="1934" />

**Comment by NilsJacobsen** [comment]
Well, the problem is that people are confused about which plugin to take. <br><br>**Second day (probably third day) problem:** 

* Really understand what's going on inside a plugin

I don't think that a huge percentage does have this problem, but your proposal should work for them. At least if we don't get too many APIs.

**Comment by samuelstroschein** [comment]
[https://inlang.com/documentation/plugin/api](https://inlang.com/documentation/plugin/api)

**Comment by felixhaeberle** [comment]
see also [https://discord.com/channels/897438559458430986/1084866749842870333/1247389185862533220](https://discord.com/channels/897438559458430986/1084866749842870333/1247389185862533220) where user has issue identifying how to get Sherlock to work. This happens on a weekly basis, and probably much more silently.

**Comment by NilsJacobsen** [comment]
What module is your create project button adding?

**Comment by NilsJacobsen** [comment]
So they add it that way and then are stuck if they want to do something else than messageFormat?

**Comment by NilsJacobsen** [comment]
Besides that (we agreed on adding it -> moved to TODO), how many users use the create button? Is it the legacy users that get into this problem? Why are there that many day-two-problem users? Best case they don't have to think about the issue until they are pro users.
